server/src/features/users/user.controller.js

```diff
@@ -128,4 +130,195 @@ export default class UserController {
     }
   }
+
+  /**
+   * Toggle user block status
+   * PATCH /api/users/:userId/block-status
+   * Body: { action: 'block' | 'unblock' }
+   */
+  static async toggleBlockStatus(req, res, next) {
+    try {
+      const { userId } = req.params;
+      const { action } = req.body;
+
+      // Validate request
+      const { error } = toggleBlockUserSchema.validate({ userId, action });
+      if (error) {
+        return res.status(400).json({
+          success: false,
+          message: error.details[0].message,
+        });
+      }
+
+      // Call service to handle the block/unblock logic
+      const result = await UserService.toggleBlockStatus(
+        req.user._id, // current admin's ID
+        userId,       // target user's ID
+        action        // 'block' or 'unblock'
+      );
+
+      return res.status(200).json({
+        success: true,
+        message: result.message,
+        data: {
+          userId: result.userId,
+          status: result.status,
+        },
+      });
+    } catch (err) {
+      // Pass the error to the error handling middleware
+      next(err);
+    }
+  }
+
+  /**
+   * Add an event to user's favorites
+   * POST /api/users/favorites
+   * Body: { eventId: string }
+   */
+  static async addToFavorites(req, res, next) {
+    try {
+      const { eventId } = req.body;
+      const userId = req.user._id;
+
+      // Validate request
+      const { error } = favoriteEventSchema.validate({ eventId });
+      if (error) {
+        return res.status(400).json({
+          success: false,
+          message: error.details[0].message,
+        });
+      }
+
+      // Check if user is allowed to add to favorites
+      const allowedRoles = ['student', 'staff', 'ta', 'professor'];
+      if (!allowedRoles.includes(req.user.role)) {
+        return res.status(403).json({
+          success: false,
+          message: 'Only students, staff, TAs, and professors can add events to favorites',
+        });
+      }
+
+      // Check if event exists
+      const event = await mongoose.model('Event').findById(eventId);
+      if (!event) {
+        return res.status(404).json({
+          success: false,
+          message: 'Event not found',
+        });
+      }
+
+      // Add to favorites if not already added
+      const user = await User.findById(userId);
+      if (!user.favoriteEvents.includes(eventId)) {
+        user.favoriteEvents.push(eventId);
+        await user.save();
+      }
+
+      return res.status(200).json({
+        success: true,
+        message: 'Event added to favorites',
+        data: {
+          eventId,
+          favoritesCount: user.favoriteEvents.length,
+        },
+      });
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  /**
+   * Remove an event from user's favorites
+   * DELETE /api/users/favorites/:eventId
+   */
+  static async removeFromFavorites(req, res, next) {
+    try {
+      const { eventId } = req.params;
+      const userId = req.user._id;
+
```

Comment:
```suggestion
// Check event exists
const eventExists = await mongoose.model('Event').findOne({
  _id: eventId,
  deletedAt: null
});

if (!eventExists) {
  return res.status(404).json({
    success: false,
    message: 'Event not found or deleted'
  });
}
```
Currently you remove the event whether it exists or not.

Reply:
mana ezay ha remove mn el favorites, lw el event msh b exist asln msh hayban fe 7eta